include/LJ.h

```diff
@@ -78,291 +82,397 @@ class LJD // Late Joiner Data basic class
+template<class PublisherSubscriber> struct LateJoinerDataTraits
 {
 };
 
-template<> struct LJD_traits<Publisher>
+template<> struct LateJoinerDataTraits<DataWriter>
 {
     typedef PublisherAttributes Attributes;
-    typedef void (DSManager::* AddEndpoint)(Publisher *);
-    typedef Publisher * (DSManager::* GetEndpoint)(GUID_t &);
-    typedef bool(*removeEndpoint)(Publisher*);
+    typedef void (DSManager::* AddEndpoint)(
+            DataWriter*);
+    typedef DataWriter* (DSManager::* GetEndpoint)(
+            GUID_t&);
+    //typedef bool(*removeEndpoint)(DataWriter*);
+    typedef ReturnCode_t (DSManager::* removeEndpoint)(
+            DataWriter*);
 
     static const std::string endpoint_type;
     static const AddEndpoint add_endpoint_function;
     static const GetEndpoint retrieve_endpoint_function;
     static const removeEndpoint remove_endpoint_function;
 
     // we don't want yet to listen on publisher callbacks
-    static Publisher * createEndpoint(Participant* part, const PublisherAttributes&, void * = nullptr);
+    static DataWriter* createEndpoint(
+            fastdds::dds::DomainEntity* part,
+            fastdds::dds::Topic* topic,
+            const PublisherAttributes&,
+            void* = nullptr);
 };
 
-template<> struct LJD_traits<Subscriber>
+template<> struct LateJoinerDataTraits<DataReader>
 {
     typedef SubscriberAttributes Attributes;
-    typedef void (DSManager::* AddEndpoint)(Subscriber *);
-    typedef Subscriber * (DSManager::* GetEndpoint)(GUID_t &);
-    typedef bool(*removeEndpoint)(Subscriber*);
+    typedef void (DSManager::* AddEndpoint)(
+            DataReader*);
+    typedef DataReader* (DSManager::* GetEndpoint)(
+            GUID_t&);
+    //typedef bool(*removeEndpoint)(DataReader*);
+    typedef ReturnCode_t (DSManager::* removeEndpoint)(
+            DataReader*);
 
     static const std::string endpoint_type;
     static const AddEndpoint add_endpoint_function;
     static const GetEndpoint retrieve_endpoint_function;
     static const removeEndpoint remove_endpoint_function;
 
-    static Subscriber * createEndpoint(Participant* part, const SubscriberAttributes&, SubscriberListener * = nullptr);
+    static DataReader* createEndpoint(
+            fastdds::dds::DomainEntity* part,
+            fastdds::dds::Topic* topic,
+            const SubscriberAttributes&,
+            fastdds::dds::SubscriberListener* = nullptr);
 };
 
-template<class PS> class DED;
+template<class PublisherSubscriber> class DelayedEndpointDestruction;
 
-template<class PS>
-class DEC
-    : public LJD // Delayed Enpoint Creation
+template<class PublisherSubscriber>
+class DelayedEndpointCreation
+    : public LateJoinerData // Delayed Enpoint Creation
 {
-    typedef typename LJD_traits<PS>::Attributes Attributes;
-    Attributes * participant_attributes;
+    typedef typename LateJoinerDataTraits<PublisherSubscriber>::Attributes Attributes;
+    Attributes* participant_attributes;
```

Comment:
I suggest renaming this variable to endpoint_qos, because it is not properly speaking the participant QoS.